ubcpi/static/js/src/ubcpi.js

```diff
@@ -98,11 +111,15 @@ angular.module('UBCPI', ['ngSanitize', 'ngCookies'])
             // all status of the app. Passed it from backend so we have a synced status codes
             self.ALL_STATUS = data.all_status;
 
-            // Assign data based on what is submitted
-            assignData(self, data);
+            // Assign data based on what has been persisted
+            var persistedDataObject = get_data().then( function(persistedData) {
+
+                if ( persistedData.answer_original !== null ) {
+                    data = persistedData;
```

Comment:
Could you remove this assignment and use persistedData directly? I don't think data is being used anywhere else. Though this assignment override the variable data defined above. It may cause confusion in the future.